### PR TITLE
Fix pixel perfect on Panel Sprite 9-patch

### PR DESCRIPTION
**Pull request by Bouh**
This is for fix the renderer of Panel Sprite ("9-patch").

Lot of weird code have been added over time.
And lot of weird little patch with 1px here or 2px somewhere else for obscure reasons.

I've read the code and rework and verify twice on my PC, and a other user have check too.

[For see the problem check this topic on forum.](https://forum.gdevelop-app.com/t/les-9-panels-sprites-ne-saffichent-pas-correctement/21867/3)

This need to be verify with a retina screen from Apple.
I hope in future we will be able to have same quality of preview in scene editor.

### CODE REVIEW

**Comment by 4ian** [comment]
😍

> Lot of weird code have been added over time.
And lot of weird little patch with 1px here or 2px somewhere else for obscure reasons.

Yeah surely my fault ☝️ I've done a wrong fix surely at some point that generated another bad fix and so on 😬

Thanks for checking that, I'll test that tonight on a few games :)

File: Extensions/PanelSpriteObject/panelspriteruntimeobject-pixi-renderer.js [hunk 1]
```diff
@@ -1,204 +1,343 @@
...
+
+  this._centerSprite.position.x = obj._lBorder;
+  this._centerSprite.position.y = obj._tBorder;
+
+  //Right
+  this._borderSprites[0].position.x = obj._width - obj._rBorder;
+  this._borderSprites[0].position.y = obj._tBorder;
+
+  //Top-right
+  this._borderSprites[1].position.x = obj._width - this._borderSprites[7].width ;
```
Comment (4ian) [inline Extensions/PanelSpriteObject/panelspriteruntimeobject-pixi-renderer.js]: This is equivalent, but I guess this should be `this._borderSprites[1].width`?

**Comment by 4ian** [comment]
I've tested on a few games and this seems to be perfect! For example, there was glitches in the space shooter "TilesMechanical1" object, that are now gone during the game 👍

Great work! Could we port these fixes to RenderedPanelSpriteInstance? Have you tried?

So happy to see these hacks gone for good! I believe the root of all the issues is that I started to add some +1/-1 in `setTexture`.

**Comment by Bouh** [comment]
Oh i didn't see  RenderedPanelSpriteInstance, i will take a look

**Comment by Bouh** [comment]
It's sightly better :D !

**Comment by 4ian** [comment]
Sounds good! Will merge later today or tomorrow :)

Is there a way to make some documentation update or something to make more obvious that the renderers for the objects in the scene editor are in another folder?
Ideally I would move these files to each extension folder 🧐 (though it's not as easy, as built in extensions don't have a way to expose a JS file to the IDE).

**Comment by Bouh** [comment]
In the file itself?
Or in a readme in the parent directory of the files.
With a small link on the main readme?

I didn't find it because when I look for links between the files I search through the whole codebase with the names of the methods.

**Comment by Wend1go** [comment]
This might also fix #1052
I have a test project linked in the issue.

**Comment by 4ian** [comment]
Merging this now :) Thanks a lot for these fixes! Will be great to finally have pixel perfect 9-patch objects 👏

**Pull request merged**